prelude/runtime/string.c

```diff
@@ -37,15 +37,20 @@ SkipInt SKIP_String_hash(char* obj) {
   return (SkipInt)str->hash;
 }
 
-// Allocates a string (with no data nor hash).
+// Allocates and nul-terminates a string (with no data nor hash).
 char* sk_string_alloc(uint32_t size) {
   sk_string_t* obj =
-      (sk_string_t*)SKIP_Obstack_alloc(size + sk_string_header_size);
+      (sk_string_t*)SKIP_Obstack_alloc(sk_string_header_size + size + 1);
   obj->size = size;
-  return (char*)&(obj->data);
+  char* str = (char*)&(obj->data);
```

Comment:
I think the cast is unneeded as you changed `sk_string_t` `data` field from `unsigned char` to `char` in a previous PR

Reply:
There is still the distinction between pointers and arrays: `error: incompatible pointer types initializing 'char *' with an expression of type 'char (*)[0]'`

Reply:
Use `&obj->data[0]`?

Reply:
Oh, yeah, nice!